sdk/core/core/test/cmocka/test_span.c

```diff
@@ -288,6 +296,56 @@ static void az_span_find_error_cases_fail()
   assert_int_equal(az_span_find(AZ_SPAN_NULL, target), -1);
 }
 
+static void az_span_find_target_longer_than_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("aa");
+  az_span target = AZ_SPAN_FROM_STR("aaa");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_target_overlap_continuation_of_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd");
+  az_span target = AZ_SPAN_FROM_STR("cde");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_target_more_chars_than_preffix_of_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd");
+  az_span target = AZ_SPAN_FROM_STR("zab");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_overlaping_target_success()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcdefghij");
+  az_span target = az_span_slice(span, 6, 9);
+
+  assert_int_equal(az_span_find(span, target), 6);
+}
+
+static void az_span_find_embedded_NULLs_success()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd\0\0fghij");
+  az_span target = AZ_SPAN_FROM_STR("\0\0");
+
+  assert_int_equal(az_span_find(span, target), 4);
+}
+
+static void az_span_find_capacity_checks_success()
+{
+  uint8_t* buffer = (uint8_t*)"aaaa";
+
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 4), az_span_init(buffer, 2, 3)), 0);
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 3), az_span_init(buffer, 2, 4)), 0);
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 3), az_span_init(buffer, 0, 2)), 0);
```

Comment:
Same issue here, I think.

Reply:
Az_span_init is: az_span_init(ptr, length, capacity).

So az_span_init(buffer, 2, 3) will produce "aa" (source), with capacity of 3.
And az_span_init(buffer, 0, 2) will produce "" (target), with capacity of 2.
So target will still be contained in source, at position 0 of source, according to your comment [here](https://github.com/Azure/azure-sdk-for-c/pull/541#discussion_r402583720):

"Here is what I expect the results to be:
az_span_find(empty, empty) = 0
**az_span_find(non-empty, empty) = 0**
az_span_find(non-empty, non-empty) = 0
az_span_find(empty, non-empty) = -1"

Reply:
Yep, resolving.